### PR TITLE
add ability to use optimisticScenario

### PR DESCRIPTION
**Pull request by dmtsolovey**

### CODE REVIEW

File: template.js [hunk 1]
```diff
@@ -69,11 +69,12 @@ sendHttpRequest(data.url, (statusCode, headers, body) => {
...
-    if (statusCode >= 200 && statusCode < 300) {
-        data.gtmOnSuccess();
-    } else {
-        data.gtmOnFailure();
+    if (!data.useOptimisticScenario) {
+        if (statusCode >= 200 && statusCode < 300) {
+            data.gtmOnSuccess();
+        } else {
+            data.gtmOnFailure();
+        }
```
Comment (kHorozhanov) [inline template.js:77]: There is no call for `data.gtmOnSuccess()` when useOptimisticScenario is turned on

**Review by kHorozhanov — CHANGES_REQUESTED** [review]
There is no call for data.gtmOnSuccess() when useOptimisticScenario is turned on

**Comment by gvidoou** [comment]
@kHorozhanov check it pls

**Review by Bukashk0zzz — APPROVED** [review]

**Review by kHorozhanov — CHANGES_REQUESTED** [review]
Call `gtmOnSuccess` immediately after sending a request

**Review by kHorozhanov — APPROVED** [review]

**Pull request merged**